File: 1518-water-bottles/1518-water-bottles.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int result = numBottles;

        while (numBottles >= numExchange) {
            int empty = numBottles % numExchange;
            numBottles = numBottles / numExchange;
            result += numBottles;
            numBottles += empty;
        }

        return result;
    }
}

File: 1518-water-bottles/1518-water-bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        result = numBottles

        while numBottles >= numExchange:
            empty = numBottles % numExchange
            numBottles = numBottles // numExchange
            result += numBottles
            numBottles += empty

        return result